tests/integration/test_linspace.py

```diff
@@ -15,46 +15,307 @@
 
 import numpy as np
 import pytest
+from utils.generators import broadcasts_to, mk_seq_array
 
 import cunumeric as num
 
 
-def test_basic():
-    x = np.linspace(2.0, 3.0, num=5)
-    y = num.linspace(2.0, 3.0, num=5)
+def equivalent_shapes_gen(shape):
+    """
+    Generate more equivalent shapes by removing
+    leading singleton dimensions from `shape`.
+    e.g., shape=(1, 4, 1), yield (1, 4, 1), (4, 1)
+    shape=(1, 1, 5), yield (1, 1, 5), (1, 5), (5,)
+    """
+    yield shape
+    for i in range(len(shape) - 1):
+        if shape[i] == 1:
+            i += 1
+            yield shape[i:]
+        else:
+            break
+
+
+@pytest.mark.parametrize(
+    "endpoint", (True, False), ids=lambda endpoint: f"(endpoint={endpoint})"
+)
+@pytest.mark.parametrize(
+    "number", (0, 1, 10), ids=lambda number: f"(num={number})"
+)
+@pytest.mark.parametrize(
+    "values",
+    ((10, -5.5), (2.0, 3.0), (0, 0), (1 + 2.5j, 10 + 5j), (0j, 10)),
+    ids=lambda values: f"(values={values})",
+)
+def test_scalar_basic(values, number, endpoint):
+    start, stop = values
+    x = np.linspace(start, stop, num=number, endpoint=endpoint)
+    y = num.linspace(start, stop, num=number, endpoint=endpoint)
     assert np.array_equal(x, y)
 
 
-def test_endpoint():
-    x = np.linspace(2.0, 3.0, num=5, endpoint=False)
-    y = num.linspace(2.0, 3.0, num=5, endpoint=False)
+@pytest.mark.parametrize(
+    "endpoint", (True, False), ids=lambda endpoint: f"(endpoint={endpoint})"
+)
+@pytest.mark.parametrize(
+    "number", (0, 1, 10), ids=lambda number: f"(num={number})"
+)
+@pytest.mark.parametrize(
+    "values",
+    ((10, -5.5), (2.0, 3.0), (0, 0), (1 + 2.5j, 10 + 5j), (0j, 10)),
+    ids=lambda values: f"(values={values})",
+)
+def test_scalar_basic_retstep(values, number, endpoint):
+    start, stop = values
+    x = np.linspace(start, stop, num=number, endpoint=endpoint, retstep=True)
+    y = num.linspace(start, stop, num=number, endpoint=endpoint, retstep=True)
+
+    assert np.array_equal(x[0], y[0])
+    if np.isnan(x[1]) and np.isnan(y[1]):
+        pass
+    else:
+        assert x[1] == y[1]
```

Comment:
It's preferable to invert the condition if the "if" block is just an empty pass. I think this could also be written:
```suggestion
    if not np.isnan(x[1] + y[1]):
        assert x[1] == y[1]
```
(since nan values are "contagious")

But also: what should happen if one is nan and the other is not? Is that a failure?

Reply:
I prefer 
`if not (np.isnan(x[1]) and np.isnan(y[1])):`
In my way, if one is nan and the other is not, it would report AssertionError when `assert x[1] == y[1]`